packages/nameguard-js/src/graphemes.ts

```diff
@@ -1,44 +1,578 @@
 import { splitGraphemes as splitGraphemesInternal } from "text-segmentation";
+import { splitCharacters, isEmoji } from "./utils";
 
-// for Hangul Jamo splitting
-const HANGUL_REGEXP = new RegExp(/[ᄀ-ᇿ]/.source, "gu");
-// TODO: replace with unicode character range instead of those pure symbols
-// [\u1100–\u11FF]
+const HANGUL_JAMO = new Set([
```

Comment:
A few ideas here:

Could we please document:

1. Why we've created this `Set`. Ex: Why are we doing something special for Hangul Jamo?
2. How we determined the values of this set. Ex: through what data source / process did we produce these values?

Also, what do you think about moving these values into a ".json" file under the `/data` directory? I think it's nice if we consolidate ideas like this together.

Lastly, what do you think about abstracting this idea a bit more? For example, as I understand, each of these values will trigger special grapheme splitting rules. Currently it might be that we're only putting Hangul Jamo values into this special set, but do you think we might ever put something other than Hangul Jamo into this set? If so, maybe we define this set more abstractly? Perhaps it just happens to be that for now only Hangul Jamo values are put into this set.

Appreciate your advice 😄 !

Reply:
Jamo are very special, they even have their own data files in the unicode standard and special handling logic. I'm worried about over-abstracting here, this logic is made specifically for jamo, it relies on the way jamo are combined into single graphemes and how Unicode NFC normalization replaces these graphemes with dedicated codepoints. However, arbitrary jamo can be combined (into graphemes which do not have dedicated codepoints) and that's the reason this custom logic exists. Fonts will render these graphemes as visually separate glyphs, because they lack the ability to render arbitrary jamo sequences. Korean people can read them though because they know the jamo composition rules.

I'll move this into a json.